### PR TITLE
Add organization search form

### LINKED ISSUE #237: Add a global search bar
Dans la page d'accueil des structures, il faudrait ajouter une barre de recherche qui permet de chercher :

- les utilisateurs par nom, prénom, nivol, mail, phone (dans son organisation et ses enfants)
- les véhicules par nom (dans son organisation et ses enfants)
- les organizations enfant par nom

La recherche doit afficher les résultats tout en affichant à nouveau le champ de recherche pré-rempli.

Si on recherche "Doe John", la requête sql doit être : `WHERE (user.firstname ILIKE ('%doe%') OR user.firstname ILIKE ('%doe%') ...) AND  (user.firstname ILIKE ('%john%') OR user.firstname ILIKE ('%john%') ...)` 

Les résultats doivent contenir des liens vers les pages correspondantes aux différentes entités.

### PR DESCRIPTION
**Pull request by vincentchalamon**
Fixes #237

I didn't see the interest of using Symfony form here.

### CODE REVIEW

File: src/Repository/CommissionableAssetRepository.php [hunk 1]
```diff
@@ -26,6 +26,20 @@ public function __construct(ManagerRegistry $registry)
         parent::__construct($registry, CommissionableAsset::class);
     }
 
+    /**
+     * @return CommissionableAsset[]
+     */
+    public function search(string $query): array
+    {
+        $words = explode(' ', $query);
```
Comment (mRoca) [inline src/Repository/CommissionableAssetRepository.php:34]: you can do a `preg_replace('/\s+/', ' ', trim($query));` before exploding the string

File: src/Repository/CommissionableAssetRepository.php [hunk 2]
```diff
@@ -26,6 +26,20 @@ public function __construct(ManagerRegistry $registry)
...
+        $qb = $this->createQueryBuilder('ca')->innerJoin('ca.organization', 'o');
+        foreach ($words as $i => $word) {
+            $qb->andWhere("ca.name LIKE ?$i OR ca.contact LIKE ?$i OR o.name LIKE ?$i")->setParameter($i, "%$word%");
```
Comment (mRoca) [inline src/Repository/CommissionableAssetRepository.php:37]: this search is not case insensitive. You should use `ILIKE` and add proper LOWER indexes : https://api-platform.com/docs/core/performance/#search-filter

In migrations : 
```php
        $this->addSql('CREATE INDEX user_firstname_lower_idx ON users (LOWER(firstname))');
```

File: src/Migrations/Version20200409125719.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        $this->abortIf('postgresql' !== $this->connection->getDatabasePlatform()->getName(), 'Migration can only be executed safely on \'postgresql\'.');
+
+        $this->addSql('CREATE INDEX user_first_name_lower_idx ON users (LOWER(first_name))');
+        $this->addSql('CREATE INDEX user_last_name_lower_idx ON users (LOWER(last_name))');
+        $this->addSql('CREATE INDEX user_email_address_lower_idx ON users (LOWER(email_address))');
+        $this->addSql('CREATE INDEX user_identification_number_lower_idx ON users (LOWER(identification_number))');
+        $this->addSql('CREATE INDEX commissionable_asset_name_lower_idx ON commissionable_asset (LOWER(name))');
+        $this->addSql('CREATE INDEX commissionable_asset_contact_lower_idx ON commissionable_asset (LOWER(contact))');
+        $this->addSql('CREATE INDEX organization_name_lower_idx ON organization (LOWER(name))');
```
Comment (fmeynard) [inline src/Migrations/Version20200409125719.php:27]: We don't the scale that we are targeting with this project, but for high volume these indexes could cause degraded performances on the users table. Basically for every upserts we will have 4 upserts in addition ... 

My knowledge is limited on PostgreSQL, but instead i would try something like : 


```suggestion
        $this->addSql('CREATE INDEX user_searchON people ((LOWER(first_name)|| '  ' || LOWER(last_name) || '  ' || LOWER(email_address)  || '  ' || LOWER(identification_number) ));');
```

Comment (vincentchalamon) [reply src/Migrations/Version20200409125719.php:27]: That would be a great improvement, could you open a PR to fix it in a new migration please (to ensure no BC break)?

**Pull request merged**